scripts/bundle-package.ts

```diff
@@ -21,11 +21,19 @@ interface Options {
   watch?: boolean;
 }
 
+const makeExternalPredicate = (externals: string[]) => {
+  if (externals.length === 0) {
+    return () => false;
+  }
+  const pattern = new RegExp(`^(${externals.join('|')})($|/)`);
+  return (id: string) => pattern.test(id);
+};
```

Comment:
this returns a `fn`? 

I guess rollup supports this?
Why is this better?

Reply:
> this returns a fn?

yes, this returns a predicate function testing if a particular import source should be treated as external or not

> I guess rollup supports this?

yes

> Why is this better?

https://github.com/storybookjs/storybook/pull/18015#discussion_r854137308 , basically without this you have such a situation:
```js
import _ from 'lodash' // treated as external
import pickBy from 'lodash/pickBy' // bundled in your script
```

The `external` match is *exact* so anything that you import using a "deep import" will be bundled (by default). This kind of predicate fixes the problem.